source/simulator/assembly.cc

```diff
@@ -1014,6 +1014,9 @@ namespace aspect
 
         const double eta = scratch.material_model_outputs.viscosities[q];
 
+        const SymmetricTensor<4,dim> constitutive_tensor =
+          scratch.material_model_outputs.consitutive_tensors[q];
```

Comment:
I'd suggest making this a reference, rather than just copy the tensor. A rank-4 tensor in dim=3 has 6^2=36 elements, which requires 288 bytes to be copied. I think a reference may be the better way to do this.